src/main/java/Game.java

```diff
@@ -162,4 +172,134 @@ private Position getPositionFromUser() {
         }
     }
 
+    @NotNull
+    private Player getTheWinner() {
+        for (Map.Entry<Position, Cell> entry : board) {
+            Position currentPosition = entry.getKey();
+            checkFreedomLine(currentPosition);
+        }
+        if(blackScore.getNumberOfFreedomLines() > whiteScore.getNumberOfFreedomLines()) {
+            return blackPlayer;
+        }
+        return whitePlayer;
+    }
+
+
+    private void checkFreedomLine(@NotNull Position position) {
+        Stone.Color color = board.getStone(position).getColor();
+        if (countHorizontally(position, 1) == 4) {
+            if(color == Stone.Color.BLACK) {
+                blackScore.incrementNumberOfFreedomLines();
+            } else {
+                whiteScore.incrementNumberOfFreedomLines();
+            }
+        }
+        if (countVertically(position, 1) == 4) {
+            if(color == Stone.Color.BLACK) {
+                blackScore.incrementNumberOfFreedomLines();
+            } else {
+                whiteScore.incrementNumberOfFreedomLines();
+            }
+        }
+        if (countDiagonallyRight(position, 1) == 4) {
+            if(color == Stone.Color.BLACK) {
+                blackScore.incrementNumberOfFreedomLines();
+            } else {
+                whiteScore.incrementNumberOfFreedomLines();
+            }
+        }
+        if (countDiagonallyLeft(position, 1) == 4) {
+            if(color == Stone.Color.BLACK) {
+                blackScore.incrementNumberOfFreedomLines();
+            } else {
+                whiteScore.incrementNumberOfFreedomLines();
+            }
+        }
+    }
+
+    private int countVertically(@NotNull Position currentPosition, int currentStoneCount) {
```

Comment:
This system doesn't take into account positions already taken by other freedom lines, does it?

Reply:
This is specific method doesn't, but the upper method will check if the line is already taken